### PR TITLE
feat: handle error states

### PR DESCRIPTION
**Pull request by sideround**
## Description
- Throw an error if ciphertext cannot be deleted
- Throw an error if ciphertext cannot be added
- Double-check how many secrets are currently stored in the secure enclave before generating one

### CODE REVIEW

File: Sources/Keychain/secretkey/KeyEncrypting.swift [hunk 1]
```diff
@@ -120,4 +123,30 @@ public final class KeyEncrypting: KeyEncryptable {
...
+        // 1. SecItemCopyMatching will query how many secrets are currently stored in the secure enclave
+        var secretRef: CFTypeRef?
+        let status = security.SecItemCopyMatching(
+            query as CFDictionary,
+            &secretRef
+        )
+
+        // 2. Return how many secrets were retrieved from the secure enclave
+        if status == noErr, let secrets = secretRef as? [SecKey] {
+            return secrets.count
```
Comment (thomas-waite) [inline Sources/Keychain/secretkey/KeyEncrypting.swift:146]: Why not always just return here? I think this is saying if there's no error, then return the number of secrets. 

Why also have `return 0` on line 150?

File: Sources/Keychain/storage/KeyStoring.swift [hunk 2]
```diff
@@ -19,9 +19,9 @@ public final class KeyStorage: KeyStoring {
     }
 
     @discardableResult
-    public func set(data: Data, key: String) throws -> OSStatus {
+    public func set(data: Data, key: String) -> OSStatus {
         // 1. Delete any existing key before saving it
-        try delete(key: key)
+        delete(key: key)
```
Comment (thomas-waite) [inline Sources/Keychain/storage/KeyStoring.swift]: And then if this throws an error, we catch that and show a useful error to the user? We should direct them to contact support - perhaps with a link to the telegram channel or us

**Review by thomas-waite — COMMENTED** [review]
Looking good, a couple of questions

File: Sources/Keychain/secretkey/KeyEncrypting.swift (hunk 1 above)
Comment (sideround) [reply Sources/Keychain/secretkey/KeyEncrypting.swift:146]: the keychain returns an optional value, secretRef in our case - we return 0 if our output cannot be cast to an array

File: Sources/Keychain/storage/KeyStoring.swift (hunk 2 above)
Comment (sideround) [reply Sources/Keychain/storage/KeyStoring.swift]: https://github.com/dawnwallet/dawn-key-management/pull/7/commits/3b99223e940ad89ccc403ef0d89da93ee009d97a

File: Sources/Keychain/secretkey/KeyEncrypting.swift (hunk 1 above)
Comment (thomas-waite) [reply Sources/Keychain/secretkey/KeyEncrypting.swift:146]: Okay. In TypeScript/Python you'd check if `secretRef` was undefined in that case and return `0` if it was. Maybe a different standard in Swift - this works too

**Review by thomas-waite — APPROVED** [review]

**Pull request merged**